war/src/main/webapp/scripts/hudson-behavior.js

```diff
@@ -878,18 +878,48 @@ function rowvgStartEachRow(recursive,f) {
     Behaviour.specify("INPUT.required", "input-required", ++p, function(e) { registerRegexpValidator(e,/./,"Field is required"); });
 
     // validate form values to be an integer
-    Behaviour.specify("INPUT.number", "input-number", ++p, function(e) { registerRegexpValidator(e,/^(\d+|)$/,"Not an integer"); });
-    Behaviour.specify("INPUT.number-required", "input-number-required", ++p, function(e) { registerRegexpValidator(e,/^\-?(\d+)$/,"Not an integer"); });
+    Behaviour.specify("INPUT.number", "input-number", ++p, function(e) {
+        e.addEventListener('keypress', function (event) {
+            if (event.which === 69 || event.which === 101) { // Prevent user input 'e' or 'E'
+                event.preventDefault();
+            }
+        })
+        registerRegexpValidator(e,/^(\d+|)$/,"Not an integer");
+    });
+
+    Behaviour.specify("INPUT.number-required", "input-number-required", ++p, function(e) {
+        e.addEventListener('keypress', function (event) {
+            if (event.which === 69 || event.which === 101) { // Prevent user input 'e' or 'E'
+                event.preventDefault();
+            }
+        })
+        registerRegexpValidator(e,/^\-?(\d+)$/,"Not an integer");
+    });
 
     Behaviour.specify("INPUT.non-negative-number-required", "input-non-negative-number-required", ++p, function(e) {
+        e.addEventListener('keypress', function (event) {
+            if (event.which === 69 || event.which === 101) { // Prevent user input 'e' or 'E'
+                event.preventDefault();
+            }
+        })
         registerRegexpValidator(e,/^\d+$/,"Not a non-negative number");
     });
 
     Behaviour.specify("INPUT.positive-number", "input-positive-number", ++p, function(e) {
+        e.addEventListener('keypress', function (event) {
+            if (event.which === 69 || event.which === 101) { // Prevent user input 'e' or 'E'
+                event.preventDefault();
+            }
+        })
         registerRegexpValidator(e,/^(\d*[1-9]\d*|)$/,"Not a positive integer");
     });
 
     Behaviour.specify("INPUT.positive-number-required", "input-positive-number-required", ++p, function(e) {
+        e.addEventListener('keypress', function (event) {
+            if (event.which === 69 || event.which === 101) { // Prevent user input 'e', 'E', '.'
```

Comment:
The `'.'` is not included in the `if` clause

Reply:
I forgot to add it. I feel so sorry.
By the way, is it a good idea to add `'.'` constraint for all input validator? Because some error message shows "Not an **integer**" and other shows "Not a non-negative **number**".
I'm thinking that should I add the `'.'` constraint for the messages which contain '**integer**', or change the messages which contain '**number**' to '**integer**'?